programs/clearing_house/src/math/repeg.rs

```diff
@@ -529,6 +623,43 @@ mod test {
         assert_eq!(new_peg, 1000);
     }
 
+    #[test]
+    fn calculate_optimal_peg_and_budget_test() {
```

Comment:
can you check that this leads to the spread being the max spread? Should also probably have a test for the case where `target_price_gap` is positive and negative

Reply:
already existed to test max spread enforcement with positive `target_price_gap`

added a test for negative `target_price_gap`